### PR TITLE
Add title to filter

### PR DESCRIPTION
**Pull request by MortalFlesh**
`Title` is needed for preparing values (future changes).

### CODE REVIEW

**Comment by OndraM** [comment]
Isn't the title always mapped to the operator? Ie. couldn't you always deduce the title based on the value of the operator?
- `=` => `eq`
- `>` => `gt`
- `>=` => `gte`
- ...

**Comment by MortalFlesh** [comment]
Well in the `QueryParameterParser` it is always mapped like you said `=` => `eq` etc. But it is not mandatory to use `QueryParameterParser` to parse `Filters` and you can name filters as you want to.

Name will be used as part of the prepared value name.
```sql
SELECT * FROM table WHERE column = :column_eq -- :{column_name}_{filter_title}
```
and prepared values would be `['column_eq' => 'value']`

File: src/Filter/FilterWithOperator.php [hunk 1]
```diff
@@ -9,9 +9,9 @@ class FilterWithOperator extends AbstractFilter
     /** @var string */
     private $operator;
 
-    public function __construct(string $column, Value $value, string $operator)
+    public function __construct(string $column, Value $value, string $operator, string $title)
```
Comment (OndraM) [inline src/Filter/FilterWithOperator.php:12]: Any reason for different parameter order than in AbstractFilter?

**Comment by OndraM** [comment]
@MortalFlesh Well if the purpose is just a name in prepared statment, why not to generate something eg. random, or number sequence etc.? Why bother about the name? Why this should be done by hand? Or am I missing some other purpose of the title?

File: src/Filter/FilterWithOperator.php (hunk 1 above)
Comment (MortalFlesh) [reply src/Filter/FilterWithOperator.php:12]: Well actually there is none.. I did not care about it.. should I unify this?

**Comment by MortalFlesh** [comment]
Now the purpose of `title` is just for name in prepared statement, but it could change in the future.. 
And I found easier aproach to define specific title than generate something, which would add unnecessary complexity.

**Comment by OndraM** [comment]
> *Could* change in future

So then add it once this will actually change, right?

> unnecessary complexity

I don't see any complexity there - expect manually defining what looks like a redundant and useless information :). You don't even need to use named prepared parameters (ie. you don't need to name them all), you can use the  question-mark syntax, and because you are assembling the query on your own, you can safely depend on order of the parameters.

**Comment by MortalFlesh** [comment]
Named parameters in prepared statements are imho better for debuging and/or other mapping/filtering, ... which would client need to do.

**Pull request merged**